src/cli/syncback.rs

```diff
@@ -0,0 +1,320 @@
+use std::{
+    io::{self, BufReader, Write as _},
+    mem::forget,
+    path::{Path, PathBuf},
+    time::Instant,
+};
+
+use anyhow::Context;
+use clap::Parser;
+use fs_err::File;
+use memofs::Vfs;
+use rbx_dom_weak::{InstanceBuilder, WeakDom};
+use rbx_reflection::ReflectionDatabase;
+use termcolor::{BufferWriter, Color, ColorChoice, ColorSpec, WriteColor};
+
+use crate::{
+    serve_session::ServeSession,
+    syncback::{syncback_loop, FsSnapshot},
+};
+
+use super::{resolve_path, GlobalOptions};
+
+const UNKNOWN_INPUT_KIND_ERR: &str = "Could not detect what kind of file was inputted. \
+                                       Expected input file to end in .rbxl, .rbxlx, .rbxm, or .rbxmx.";
+
+/// Performs syncback for a project file
```

Comment:
This is a bit of a circular definition. It would be good to explain what a "syncback" even if it's a short description.

Reply:
Good callout. I've written some more helpful docs for the command.